activemodel/lib/active_model/type/date.rb

```diff
@@ -52,8 +52,8 @@ def fast_string_to_date(string)
           end
         end
 
-        def fallback_string_to_date(string)
-          new_date(*::Date._parse(string, false).values_at(:year, :mon, :mday))
+        def fallback_string_to_date(string, limit: 128)
+          new_date(*::Date._parse(string, false, limit: limit).values_at(:year, :mon, :mday)) if string.length <= limit
```

Comment:
Since strings longer than the limit are likely a rare mistake, what do you think about simply rescuing the error?  e.g.:

```suggestion
        def fallback_string_to_date(string)
          parts = begin
            ::Date._parse(string, false)
          rescue ArgumentError
          end

          new_date(*parts.values_at(:year, :mon, :mday)) if parts
```

It's more verbose, but it doesn't tie us to a specific limit.  So if Ruby changes the contract of `Date._parse` again in the future, we may be covered.

Reply:
I'm not sure I'm comfortable rescuing generic exceptions like `ArgumentError`, even though this pattern is popular in the rails codebase. What do you think about at least matching an error message?

This is not very probably that the ruby will change `_parse` method signature, but in this case suggested approach  will potentially also have problems and we will get `ArgumentError` for another reason than we originally intended.

Reply:
I am also a bit wary of rescuing `ArgumentError`.  But my thinking was: if Ruby reduces the set of accepted inputs again (whether by changing the limit or adding some other default argument), then our response would be same -- we would prevent the error from bubbling up, and return nil.

I feel like matching on the error message may be too fragile (more so than rescuing `ArgumentError`).